source/kraken/tests/apply_disruption_test.cpp

```diff
@@ -109,3 +117,855 @@ BOOST_FIXTURE_TEST_CASE(simple_train_cancellation, SimpleDataset) {
+                              .on(nt::Type_e::StopPoint, "stop3")
+                              // 2012/6/14 7h -> 2012/6/17 6h
+                              .application_periods(btp("20120614T070000"_dt, "20120617T060000"_dt))
+                              // 2012/6/17 23h -> 2012/6/18 6h
+                              .application_periods(btp("20120617T230000"_dt, "20120618T060000"_dt))
+                              // 2012/6/19 4h -> 2012/6/19 8h30
+                              .application_periods(btp("20120619T083000"_dt, "20120619T093000"_dt))
+                              .get_disruption(),
+                              *b.data->pt_data, *b.data->meta);
+
+    // Base VJ
+    auto* vj  = b.data->pt_data->vehicle_journeys_map["vj:1"];
+    BOOST_CHECK_MESSAGE(ba::ends_with(vj->base_validity_pattern()->days.to_string(), "111111"),
+            vj->base_validity_pattern()->days);
+    BOOST_CHECK_MESSAGE(ba::ends_with(vj->adapted_validity_pattern()->days.to_string(), "011000"),
+            vj->adapted_validity_pattern()->days);
+
+    // Adapted VJ
+    vj  = b.data->pt_data->vehicle_journeys_map["vj:1:Adapted:0:stop3_closed"];
+    BOOST_CHECK_MESSAGE(ba::ends_with(vj->base_validity_pattern()->days.to_string(), "000000"),
+            vj->base_validity_pattern()->days);
+    BOOST_CHECK_MESSAGE(ba::ends_with(vj->adapted_validity_pattern()->days.to_string(), "100111"),
+            vj->adapted_validity_pattern()->days);
+
+    b.data->build_uri();
+    b.data->pt_data->index();
+    b.finish();
+    b.data->build_raptor();
+    navitia::routing::RAPTOR raptor(*(b.data));
+    auto compute = [&](nt::RTLevel level, const std::string& from, const std::string& to,
+            navitia::DateTime datetime, int day) {
+        return raptor.compute(b.data->pt_data->stop_areas_map.at(from), b.data->pt_data->stop_areas_map.at(to),
+                datetime, day, navitia::DateTimeUtils::inf, level, 2_min, true);
+    };
+
+    // 2012/6/15 8:00
+    auto res = compute(nt::RTLevel::Base, "stop1", "stop3", "08:00"_t, 1);
+    BOOST_REQUIRE_EQUAL(res.size(), 1);
+    res = compute(nt::RTLevel::RealTime, "stop1", "stop4", "08:00"_t, 1);
+    BOOST_REQUIRE_EQUAL(res.size(), 1);
+    res = compute(nt::RTLevel::RealTime, "stop1", "stop3", "08:00"_t, 1);
+    BOOST_REQUIRE_EQUAL(res.size(), 0); // <-- The stop3 is not accessible at the given datetime.
+
+    // 2012/6/18 5:00
+    res = compute(nt::RTLevel::Base, "stop1", "stop3", "05:00"_t, 4);
+    BOOST_REQUIRE_EQUAL(res.size(), 1);
+    res = compute(nt::RTLevel::RealTime, "stop1", "stop3", "05:00"_t, 4);
+    BOOST_REQUIRE_EQUAL(res.size(), 1); // <-- The result should be the base vj
+    BOOST_REQUIRE_EQUAL(res[0].items[0].stop_times[0]->vehicle_journey->uri, "vj:1");
+
+    navitia::delete_disruption("stop3_closed", *b.data->pt_data, *b.data->meta);
+
+    check_vjs_without_disruptions(b.data->pt_data->vehicle_journeys);
+
```

Comment:
minor: same as before, I think it would be nice to do again some journey computation after the deletion (you can copy/paste:

```
BOOST_REQUIRE_EQUAL(compute(nt::RTLevel::RealTime, "stop1", "stop3", "08:00"_t, 1).size(), 1)
BOOST_REQUIRE_EQUAL(compute(nt::RTLevel::RealTime, "stop1", "stop3", "05:00"_t, 4).size(), 1)
```